lib/cc/engine/scanner_config.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module CC
+  module Engine
+    class ScannerConfig
+      CATEGORIES = [
+        "Bug Risk", "Clarity", "Compatibility", "Complexity", "Duplication",
+        "Performance", "Security", "Style"
+      ].freeze
+      SEVERITIES = %w[info minor major critical blocker].freeze
+
+      class InvalidConfigError < StandardError; end
+
+      def initialize(config, check_name)
+        @config = config
+        @check_name = check_name
+
+        validate_config!
+      end
+
+      def pattern
+        config.fetch("pattern")
+      end
+
+      def severity
+        config.fetch("severity", "minor")
+      end
+
+      def description
+        config.fetch("annotation")
+      end
+
+      def categories
+        Array(config.fetch("categories", "Bug Risk"))
+      end
+
+      def path_patterns
+        Array(config.fetch("path_patterns", ["**/*"]))
+      end
+
+      def content
+        config["content"]
+      end
+
+      private
+
+      attr_reader :config, :check_name
+
+      def validate_config!
+        validate_required_config_entries!
+        validate_severity!
+        validate_categories!
+      end
+
+      def validate_required_config_entries!
+        unless config.key?("pattern")
+          raise InvalidConfigError, "Pattern is missing from #{check_name}"
+        end
+        if !config.key?("annotation") || config["annotation"].strip.empty?
+          raise InvalidConfigError, "Annotation is missing from #{check_name}"
+        end
+      end
+
+      def validate_severity!
```

Comment:
> Will default validation give any info that can help fix the config?

I can't speak to exactly what the unified handling is or does, the only benefit to using it is that it'd be centralized and if/when we _did_ add pointers like that, all engines would benefit.

Anyway, this is fine to keep here since it was explicitly scoped.